security/security-spring/src/main/java/org/cbioportal/security/spring/CancerStudyPermissionEvaluator.java

```diff
@@ -0,0 +1,396 @@
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.security.spring;
+
+// imports
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.cbioportal.model.CancerStudy;
+import org.cbioportal.model.GeneticProfile;
+import org.cbioportal.model.Sample;
+import org.cbioportal.model.SampleList;
+import org.cbioportal.persistence.GeneticProfileRepository;
+import org.cbioportal.persistence.SampleRepository;
+import org.cbioportal.persistence.SampleListRepository;
+import org.cbioportal.persistence.SecurityRepository;
+import org.cbioportal.persistence.StudyRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.access.PermissionEvaluator;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.AuthorityUtils;
+
+/**
+ * A custom PermissionEvaluator implementation that checks whether a
+ * particular user has access to a particular cancer study.
+ *
+ * Anonymous users will only get access to public studies.
+ *
+ * @author Benjamin Gross
+ */
+class CancerStudyPermissionEvaluator implements PermissionEvaluator {
+
+    private static Log log = LogFactory.getLog(CancerStudyPermissionEvaluator.class);
+
+    @Autowired
+    private SecurityRepository securityRepository;
+
+    @Autowired
+    private StudyRepository studyRepository;
+
+    @Autowired
+    private GeneticProfileRepository geneticProfileRepository;
+
+    @Autowired
+    private SampleRepository sampleRepository;
+
+    @Autowired
+    private SampleListRepository sampleListRepository;
+
+    @Value("${authenticate}")
+    private String AUTHENTICATE;
+
+    @Value("${app.name}")
+    private String APP_NAME;
+    private String DEFAULT_APP_NAME = "public_portal";
+
+    @Value("${filter_groups_by_appname}")
```

Comment:
As it stands, code cannot deploy propertly without a defined property.  I would set a default value here.